Source/WebGPU/WGSL/Parser.cpp

```diff
@@ -823,6 +828,40 @@ Result<AST::CompoundStatement> Parser<Lexer>::parseCompoundStatement()
     RETURN_NODE(CompoundStatement, WTFMove(statements));
 }
 
+template<typename Lexer>
+Result<AST::IfStatement> Parser<Lexer>::parseIfStatement()
+{
+    START_PARSE();
+
+    PARSE(attributes, Attributes);
+
+    return parseIfStatementWithAttributes(WTFMove(attributes), _startOfElementPosition);
+}
+
+template<typename Lexer>
+Result<AST::IfStatement> Parser<Lexer>::parseIfStatementWithAttributes(AST::Attribute::List&& attributes, SourcePosition _startOfElementPosition)
```

Comment:
Just a nit, but if this is only called from `parseIfStatement`, why split it?

Reply:
It’s called recursively at line 854. I’ve split it because attributes are only valid on the top level if-stmt, not any that are part of an else clause.